extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplatesAnalysisBuildItem.java

```diff
@@ -40,13 +42,30 @@ public static final class TemplateAnalysis {
         // File path, e.g. hello.html or ItemResource/items.html
         public final String path;
 
+        /**
+         * Implicit parameter declarations extracted from Message Bundle methods used for the Data namespace validation.
+         */
+        public final Map<String, String> msgBundleMethodParamNameToType;
```

Comment:
I believe that we should make the message bundle params part of the `TemplateAnalysis.parameterDeclarations` instead. And we could do the same for checked templates. So that `templateAnalysis.parameterDeclarations` would be the only source of declarations for [the branch above](https://github.com/quarkusio/quarkus/pull/26218/files#diff-409ae75ca5b45c330d6f027342c63933ff169ca45a313721bad52bc174e4694eR769-R802).

WDYT?

Reply:
@mkouba I don't mind to do this. Before I do, I've some objections - implicit parameter declarations taken from methods are not TemplateNodes in a sense of current implementation - there is not origin (that has property lines etc.), there is no default. Also it has to be set before the template is parsed (to allow template param decl. to override it), so I'll have to iterate over once more then in current impl. Just confirm you prefer `parameterDeclarations` and I'll do that, it's easy.

Reply:
> implicit parameter declarations taken from methods are not TemplateNodes in a sense of current implementation - there is not origin (that has property lines etc.), there is no default

That's ok. Just return `null` for both methods.

> Also it has to be set before the template is parsed (to allow template param decl. to override it), so I'll have to iterate over once more then in current impl. Just confirm you prefer parameterDeclarations and I'll do that, it's easy.

We can collect the implicit param declarations during the analysis (before the `ParserHook` is registered) and then merge it with `template.getParameterDeclarations()` (ofc the explicit declarations would take precedence).